src/main/java/com/Crowdsourcing/loot/CrowdsourcingLoot.java

```diff
@@ -0,0 +1,208 @@
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_5) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I1) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I2) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I3) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I4) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I5))
+		{
+			return 2;
+		}
+
+		return 0;
+	}
+
+	private HashMap<String, Object> getMetadataForLoot(String name)
+	{
+		HashMap<String, Object> metadata = new HashMap<>();
+
+		// universal metadata
+		metadata.put("ringOfWealth", getRingOfWealth());  // 0=not wearing, 1=wearing uncharged, 2=wearing charged
+
+		for (Map.Entry<Integer, String> entry : VARBITS_CA.entrySet())
+		{
+			int varbitId = entry.getKey();
+			String caTier = entry.getValue();
+			if (client.getVarbitValue(varbitId) == CA_CLAIMED)
+			{
+				metadata.put(caTier, true);
+			}
+		}
+
+		for (Map.Entry<Integer, String> entry : VARBITS_CLUE_WARNINGS.entrySet())
+		{
+			int varbitId = entry.getKey();
+			String clueTier = entry.getValue();
+			if (client.getVarbitValue(varbitId) == CLUE_WARNING_ENABLED)
+			{
+				metadata.put(clueTier, true);
+			}
+		}
+
+		if (name == null)
+		{
+			return metadata;
+		}
+
+		// conditional metadata
+		switch (name)
+		{
+			case HUNTERS_LOOT_SACK_BASIC:
+			case HUNTERS_LOOT_SACK_ADEPT:
+			case HUNTERS_LOOT_SACK_EXPERT:
+			case HUNTERS_LOOT_SACK_MASTER:
+				metadata.putAll(createSkillMap(Skill.HERBLORE));
+				metadata.putAll(createSkillMap(Skill.WOODCUTTING));
+		}
+
+		return metadata;
+	}
+
+	@Subscribe
+	public void onLootReceived(LootReceived event)
+	{
+		String name = event.getName();
+		int combatLevel = event.getCombatLevel();
+		LootRecordType type = event.getType();
+
+		LootData data = new LootData(name, combatLevel, type);
+
+		for (ItemStack item: event.getItems())
+		{
+			int itemId = item.getId();
+			int quantity = item.getQuantity();
+			data.addItem(itemId, quantity);
+		}
+
+		data.setTick(client.getTickCount());
+		data.setLocation(client.getLocalPlayer().getWorldLocation());
+
+		clientThread.invokeLater(() -> {
+			HashMap<String, Object> metadata = getMetadataForLoot(name);
+			data.setMetadata(metadata);
+
+//			log.info(data.toString());
+			manager.storeEvent(data);
+		});
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		ChatMessageType type = event.getType();
+		if (type != ChatMessageType.GAMEMESSAGE && type != ChatMessageType.SPAM)
+		{
+			return;
+		}
+
+		String message = event.getMessage();
+		if (CLUE_MESSAGE.matcher(message).matches() || ROGUE_MESSAGE.equals(message))
+		{
+			LootData data = new LootData(null, -1, null);
+
+			data.setMessage(message);
+			data.setTick(client.getTickCount());
+			data.setLocation(client.getLocalPlayer().getWorldLocation());
+
+			clientThread.invokeLater(() -> {
+				HashMap<String, Object> metadata = getMetadataForLoot(null);
+				data.setMetadata(metadata);
```

Comment:
Same here, can we just throw all this in a `new LootData(...)` constructor instead of using set methods?